### ISSUE TITLE
Add support for the table code.

### ISSUE THREAD

**Issue by lartpang** [WGUNDERWOOD/tex-fmt#26]
@WGUNDERWOOD 
I was using LaTeX workshop in VSCode to write documents, and the latexindent it uses by default works great, but it's too slow. I saw that the latest version of LaTeX workshop supports tex-fmt. after learning how fast it is, I immediately tried it. But unfortunately it doesn't seem to handle table code very well. However latexindent aligns all the `&` very well.

**Comment by Kami-prog** [comment]
install this
https://bit.ly/4evXrQ6

you may need to install the c compiler

**Comment by lartpang** [comment]
@Kami-prog 

I directly use the binary file of `tex-fmt`.
What is the purpose of this C compiler of yours?

**Comment by WGUNDERWOOD** [comment]
I could be interested in adding support for aligning ampersands in tables. However some thought would have to be put in to the implementation.

**Comment by Timmyfox** [comment]
Just wanted to chime in and mention that this probably the biggest thing I'm missing right now with this formatter. Latexindent is extremely convenient with how it aligns all the `&`'s in the table and I find myself having to constantly switch back and forth to make use of missing features like this one. Hope to see it eventually implemented!

**Comment by mxwell-dev** [comment]
Some input regarding proper implementation: One "edge case" not handled well by latexindent (if used as reference behaviour) is double backslashes `\\` in the table, which are not row-terminators. E.g. when using `\makecell` from the `makecell` package.

**Comment by vicapow** [comment]
Here's a reference to the implementation in unified-latex if it helps: https://github.com/siefkenj/unified-latex/blob/main/packages/unified-latex-prettier/libs/printer/environment.ts#L148

**Comment by iwishiwasaneagle** [comment]
+1 for this feature. I've had to switch back to latex-indent for now purely because of this. Love tex-fmt for everything else but this is a must have feature.

**Comment by Enivex** [comment]
I'll add that this is hugely convenient for aligned equations as well

**Comment by luan-xiaokun** [comment]
tex-fmt is a really cool formatting tool! But yes, alignment is a must-have feature, temporarily switching back to latexindent. Hope this feature can be implemented before I stop using TeX.

**Comment by Timmyfox** [comment]
Yes pretty much, essentially the behavior same as `latexindent` provides.

Before:
```Latex
\begin{table}[htbp]
	\centering
	\caption{My table}
	\begin{tabular}{llll}
		\toprule
		This & is & a & table \\
		\midrule
		Foo  & bar & baz & qux \\
		Hello & world & example & table \\
		\bottomrule
	\end{tabular}
\end{table}
```

After:
```Latex
\begin{table}[htbp]
	\centering
	\caption{My table}
	\begin{tabular}{llll}
		\toprule
		This  & is    & a       & table \\
		\midrule
		Foo   & bar   & baz     & qux   \\
		Hello & world & example & table \\
		\bottomrule
	\end{tabular}
\end{table}
```

**Comment by luan-xiaokun** [comment]
One thing that I found not quite well formatted by latexindent is aligning & when there is multicolumn, for example:

latexindent formats as follows,
```latex
\begin{table}
    \caption{Table caption}
    \begin{tabular}{llll}
        \toprule
        \multicolumn{2}{c}{First two columns} & \multicolumn{2}{c}{Last two columns}                 \\
        Col 1                                 & Col 2                                & Col 3 & Col 4 \\
        \midrule
        1                                     & 2                                    & 3     & 4     \\
        \bottomrule
    \end{tabular}
\end{table}
```

In my opinion, the following may look better:
```latex
\begin{table}
    \caption{Table caption}
    \begin{tabular}{llll}
        \toprule
        \multicolumn{2}{c}{First two columns} & \multicolumn{2}{c}{Last two columns}\\
        Col 1            & Col 2              & Col 3 & Col 4 \\
        \midrule
        1                & 2                  & 3     & 4     \\
        \bottomrule
    \end{tabular}
\end{table}
```

Don't know if it is possible to support such feature.

Turning off line wrapping in table environment gives a better looking for me, I just need to hold Shift and then scroll from left to right to check each column, if all & are properly aligned.

**Comment by leanhdung1994** [comment]
@lartpang @Enivex 

The [latexindent V4.0](https://github.com/cmhughes/latexindent.pl/releases/tag/V4.0) is indeed very fast:
````ps
Copyright (C) Microsoft Corporation. All rights reserved.

Install the latest PowerShell for new features and improvements! https://aka.ms/PSWindows

PS P:\Personal projects\test performance of latexindent> powershell -Command "Measure-Command { latexindent '.\arXiv-1711.07597v2\paperSS122018arxivv2.tex' } | Select-Object -ExpandProperty TotalSeconds"
UTF-8 "\x95" does not map to Unicode at C:/Users/Akira/AppData/Local/Programs/MiKTeX/scripts/latexindent/LatexIndent/FileExtension.pm line 196, <$fh> line 18395.
UTF-8 "\x95" does not map to Unicode at C:/Users/Akira/AppData/Local/Programs/MiKTeX/scripts/latexindent/LatexIndent/FileExtension.pm line 196, <$fh> line 18395.
23.6224204
PS P:\Personal projects\test performance of latexindent>
````

It takes around 25 seconds on this [tex file](https://arxiv.org/src/1711.07597) with 36.192 lines.

**Comment by lartpang** [comment]
@leanhdung1994 

Thanks for the feedback!
I'm curious how tex-fmt compares to this in the same runtime environment?

**Comment by leanhdung1994** [comment]
@lartpang

Even latexindent is not as fast as Rust-based tex-fmt, I still stick with the former due to its extensive capability (which have been actively developed [since 2012](https://tex.stackexchange.com/a/71034/161639)). For example, from
````tex
I have     $   a    + b    $.


We have something.
````
latexindent gives
````tex
I have $a + b$.

We have something.
````

Thus illustrates its capability to removing redundant white space and blank lines. With V4.0, latexindent's performance is sufficient for almost all daily workloads.
